lib/SILOptimizer/Transforms/CopyForwarding.cpp

```diff
@@ -347,13 +349,10 @@ class AnalyzeForwardUse
       Oper = &CopyInst->getAllOperands()[CopyAddrInst::Src];
       return CopyInst->isTakeOfSrc();
     }
-    assert(!CopyInst->isInitializationOfDest() && "illegal reinitialization");
+    // TODO: should we check instead that there's a destroy_addr prior to this instruction?
     Oper = &CopyInst->getAllOperands()[CopyAddrInst::Dest];
     return true;
   }
-  bool visitStoreInst(StoreInst *Store) {
```

Comment:
I think this type of thing should be handled by the verifier, not this pass (correct me if that's not correct, though). Also, the following is now valid:
```
  %alloc = alloc_stack $ObjWrapper
  
  %val = load [copy] %1 : $*ObjWrapper
  store %val to [init] %alloc : $*ObjWrapper
  
  copy_addr [take] %0 to %alloc : $*ObjWrapper
```

Reply:
I'll make this another patch because it's not specific to ownership.